src/libstd/io/mod.rs

```diff
@@ -589,6 +1075,33 @@ pub trait BufRead: Read {
     /// return an error if the read bytes are not valid UTF-8. If an I/O error
     /// is encountered then `buf` may contain some bytes already read in the
     /// event that all data read so far was valid UTF-8.
+    ///
+    /// # Examples
+    ///
+    /// A locked standard input implements `BufRead`. In this example, we'll
+    /// read all of the lines from standard input. If we were to do this in
+    /// an actual project, the [`lines()`][lines] method would be easier, of
+    /// course.
+    ///
+    /// [lines]: #method.lines
+    ///
+    /// ```
+    /// use std::io;
+    /// use std::io::prelude::*;
+    ///
+    /// let stdin = io::stdin();
+    /// let mut stdin = stdin.lock();
+    /// let mut buffer = String::new();
+    ///
+    /// while let Ok(n) = stdin.read_line(&mut buffer) {
```

Comment:
Actually, to stay in line with the rest of the examples here, this may want to just `unwrap`:

```rust
while stdin.read_line(&mut buffer).unwrap() > 0 {
    // work with buffer
    // ...

    buffer.clear();
}
```